src/computability/ackermann.lean

```diff
@@ -0,0 +1,309 @@
+/-
+Copyright (c) 2022 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+
+import computability.primrec
+import tactic.linarith
+
+/-!
+# Ackermann function
+
+In this file, we define the two-argument Ackermann function `ack`. Despite having a recursive
+definition, we show that this isn't a primitive recursive function.
+
+## Main result
+
+- `not_primrec₂_ack`: the two-argument Ackermann function is not primitive recursive.
+-/
+
+namespace nat
```

Comment:
I couldn't put them in the file where `mkpair` was defined, since I wasn't able to import `linarith` there. I guess I could try proving them without linarith and moving them there.

Reply:
#15539